### PR TITLE
Bump sbt to 1.5.4

### PR DESCRIPTION
**Pull request by ignasi35**
~#354  is a direct bump from sbt 1.3.x to 1.5.1.~

~Let's try the bump in two steps?~

Going to sbt 1.4.9 first, and then 1.5.1 was no solution to prevent the errors in #354 so I've re-purposed this PR and I'm bumping sbt to 1.5.1 directly.

### CODE REVIEW

**Comment by ignasi35** [comment]
Duplicate of #347

**Comment by ignasi35** [comment]
This is failing with the error:

```
[error] scala.tools.reflect.ToolBoxError: reflective compilation has failed:
[error] 
[error] illegal cyclic reference involving type BuilderType
[error]         at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$ToolBoxGlobal.throwIfErrors(ToolBoxFactory.scala:331)
[error]         at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$ToolBoxGlobal.wrapInPackageAndCompile(ToolBoxFactory.scala:213)
[error]         at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$ToolBoxGlobal.compile(ToolBoxFactory.scala:267)
[error]         at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl.$anonfun$compile$13(ToolBoxFactory.scala:444)
[error]         at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl$withCompilerApi$.apply(ToolBoxFactory.scala:370)
[error]         at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl.compile(ToolBoxFactory.scala:437)
[error]         at scala.tools.reflect.ToolBoxFactory$ToolBoxImpl.eval(ToolBoxFactory.scala:459)
[error]         at build.play.grpc.ReflectiveCodeGen$.loadAndSetGenerator(ReflectiveCodeGen.scala:137)
[error]         at build.play.grpc.ReflectiveCodeGen$.$anonfun$projectSettings$12(ReflectiveCodeGen.scala:77)
[error]         at build.play.grpc.ReflectiveCodeGen$.$anonfun$projectSettings$12$adapted(ReflectiveCodeGen.scala:68)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
...
```

After instapecting and debugging, the compile error is on:

```
  val generators =
    AkkaGrpcPlugin.generatorsFor(sources, languages, scalaBinaryVersion, logger) ++
    Seq($extraGenerators)
  -----^
		.map(gen => 
			GeneratorBridge.sandboxedGenerator(gen, scalaBinaryVersion, akka.grpc.gen.StdoutLogger)
		)
```
 
The `extraGnerators` is a list of `object` instances where each instance is a Akka-gRPC or Play-gRPC generator. 

Surprisingly:

- the `illegal cyclic reference involving type BuilderType` appears when bumping sbt from 1.3.13 to 1.4.0
- there's not error on the equivalent code in [the Akka-gRPC build](https://github.com/akka/akka-grpc/blob/master/project/ReflectiveCodeGen.scala#L130)

**Comment by ignasi35** [comment]
This is related to https://github.com/scala/bug/issues/12038 but bumping scala on the meta build doesn't fix the issue.

**Comment by ignasi35** [comment]
More info, the error happens when building a `Seq` of code generators. Surprisingly, when the Seq contains a only one item, the execution succeeds. So, building the following project works: 

```
// this one works and the code is generated
val playTestdata = Project("play-grpc-testdata", file("play-testdata"))
  .dependsOn(playRuntime)
  .settings(
    ReflectiveCodeGen.extraGenerators ++= List(
      "akka.grpc.gen.scaladsl.ScalaMarshallersCodeGenerator", // only one generator
//      "play.grpc.gen.javadsl.PlayJavaClientCodeGenerator",
//      "play.grpc.gen.javadsl.PlayJavaServerCodeGenerator",
//      "play.grpc.gen.scaladsl.PlayScalaClientCodeGenerator",
//      "play.grpc.gen.scaladsl.PlayScalaServerCodeGenerator",
    ),
    scalacOptions += "-Xlint:-unused,_",  // can't do anything about unused things in generated code
...
```

but thisone triggers the error `illegal cyclic...`:

```
// this one works and the code is generated
val playTestdata = Project("play-grpc-testdata", file("play-testdata"))
  .dependsOn(playRuntime)
  .settings(
    ReflectiveCodeGen.extraGenerators ++= List(
      "akka.grpc.gen.scaladsl.ScalaMarshallersCodeGenerator",
      "play.grpc.gen.javadsl.PlayJavaClientCodeGenerator", // two generators on the list
//      "play.grpc.gen.javadsl.PlayJavaServerCodeGenerator",
//      "play.grpc.gen.scaladsl.PlayScalaClientCodeGenerator",
//      "play.grpc.gen.scaladsl.PlayScalaServerCodeGenerator",
    ),
    scalacOptions += "-Xlint:-unused,_",  // can't do anything about unused things in generated code
...
```

Commenting/uncommenting any single one of the five generators will result in a successful execution. It's not the value, but the number of items.

File: build.sbt [hunk 1]
```diff
@@ -91,7 +91,6 @@ val playActionsTestData = Project("play-grpc-actions-testdata", file("play-actio
     ReflectiveCodeGen.extraGenerators ++= List(
       "play.grpc.gen.scaladsl.PlayScalaServerCodeGenerator",
     ),
-    ReflectiveCodeGen.codeGeneratorSettings += "use_play_actions",
```
Comment (raboof) [inline build.sbt:94]: heh, good catch

File: build.sbt [hunk 2]
```diff
@@ -100,10 +99,11 @@ val playActionsTestData = Project("play-grpc-actions-testdata", file("play-actio
       Dependencies.Compile.akkaDiscovery,
     ),
   )
-  .pluginTestingSettings
   .enablePlugins(build.play.grpc.NoPublish)
+  .pluginTestingSettings
 
-val playGenerators = Project("play-grpc-generators", file("play-generators"))
+val playGrpcCodegenId = "play-grpc-generators"
```
Comment (raboof) [inline build.sbt:105]: Is this used anywhere?

File: play-testkit/src/test/scala/example/myapp/helloworld/grpc/actions/helloworld/GreeterServiceWithActionsImpl.scala [hunk 3]
```diff
@@ -21,8 +21,6 @@ class GreeterServiceWithActionsImpl @Inject() (
     actionBuilder: DefaultActionBuilder,
 ) extends AbstractGreeterServiceRouter(
       actorSystem,
-      parsers,
-      actionBuilder,
```
Comment (raboof) [inline play-testkit/src/test/scala/example/myapp/helloworld/grpc/actions/helloworld/GreeterServiceWithActionsImpl.scala:25]: If they don't need to be passed to AbstractGreeterServiceRouter anymore, maybe they don't need to be in the parameters for this class at all anymore?

File: project/ProjectExtensions.scala [hunk 4]
```diff
@@ -16,9 +12,7 @@ object ProjectExtensions {
     /** Add settings to test the sbt-plugin in-process */
     def pluginTestingSettings: Project =
       project
-        .enablePlugins(JavaAgent)
         .settings(
-          javaAgents += "org.mortbay.jetty.alpn" % "jetty-alpn-agent" % "2.0.10" % Test,
```
Comment (raboof) [inline project/ProjectExtensions.scala:21]: :tada:

**Review by raboof — APPROVED** [review]
Wow, that's quite a jump!

File: build.sbt (hunk 2 above)
Comment (ignasi35) [reply build.sbt:105]: nope

**Review by raboof — APPROVED** [review]
LGTM AFAICS - is the test failure a flake or a 'real' thing?

**Comment by ignasi35** [comment]
> LGTM AFAICS - is the test failure a flake or a 'real' thing?

Looks flaky considering it passed on the `Branch` build. I've relaunched the JDK11 job on the `PR` build to see what happens.

**Pull request merged**